File: stories/templates/py/{epic-verb-noun}/{sub-epic-verb-noun}/{lowest-sub-epic}/{lowest_sub_epic}_stories.py
# ---
# fidelity: [exploration, specification]
# artifact: [story-data]
# format: py
# section: leaf-spec-file
# ---
#
# Pure data only. Regeneratable, bidirectional. The adapter renders freely
# on every run and can parse this file back into the domain StoryMap.
#
# File-naming convention (Python-only exception to kebab folders):
#   Folder tree      -> {epic-verb-noun}/{sub-epic-verb-noun}/{lowest-sub-epic}/
#   Leaf spec file   -> {lowest_sub_epic}_stories.py       (snake — module import)
#   Tier file        -> {lowest_sub_epic}_<tier>.py        (snake — module import)
#   Tier test file   -> test_{lowest_sub_epic}_<tier>.py   (snake — pytest discovery)
#
# Contents rules:
#   - Module-level constant, one per Story, keyed by SCREAMING_SNAKE.
#   - Story is a plain dict; keys `story`, `actor`, `domain_terms`, `evidence`,
#     then one Scenario per additional key.
#   - Every Scenario has `name`, `given: tuple[str, ...]`, and
#     `interactions: tuple[Interaction, ...]`. Each Interaction has `when` and
#     `then`, both `tuple[str, ...]`.
#   - Step strings are plain prose. First step of each phase is unprefixed;
#     continuations start with `"And "` or `"But "` inside the string.
#   - No imports of test framework, helpers, or production code.
#   - No function definitions, no assertions.

"""Story data for the `{Sub-Epic Name}` sub-epic."""

from __future__ import annotations

from typing import Final


STORY_ONE: Final = {
    "story": "<Story Verb-Noun>",
    "actor": "<Actor>",
    "domain_terms": ("<term>", "<term>"),
    "evidence": ("<source ref>",),

    "main_flow": {
        "name": "<outcome-oriented scenario name>",
        "given": (
            "<precondition>",
            "And <continuation precondition>",
        ),
        "interactions": (
            {
                "when": ("<action>",),
                "then": (
                    "<observable outcome>",
                    "And <continuation outcome>",
                ),
            },
        ),
    },
}

File: stories/templates/js/{epic-verb-noun}/{sub-epic-verb-noun}/{lowest-sub-epic}/{lowest-sub-epic}-stories.js
// {lowest-sub-epic}-stories.js

export const BACKGROUND = [{ given: '<shared precondition>' }]

// ── Story 1 — main-flow only (Exploration fidelity) ──────────────────────────

export const VerbNounMainFlow = {
  story: '<Story Verb–Noun — main flow>',
  actor: '<Actor Name>',

  mainFlow: {
    name: '<happy-path scenario name>',
    steps: [
      { given: 'a **<ConceptA>** *<value>*' },
      { when:  'the **<actor>** *<value>* <triggering action>' },
      { then:  'the **<observed concept>** is *<observable outcome>*' },
      { and:   '<additional outcome>' },
    ],
  },
}

// ── Story 2 — inline (Specification fidelity) ─────────────────────────────────

export const VerbNounInline = {
  story: '<Story Verb–Noun — inline>',
  actor: '<Actor Name>',

  mainFlow: {
    name: '<main-flow scenario name>',
    steps: [
      { given: 'a **<ConceptA>** *<value>*' },
      { when:  'the **<actor>** *<value>* <triggering action>' },
      { then:  'the **<observed concept>** is *<observable outcome>*' },
      { and:   '<additional outcome>' },
    ],
  },

  negativePath: {
    name: '<negative scenario name>',
    steps: [
      { given: '<alternate precondition>' },
      { when:  '<alternate action>' },
      { then:  '<negative outcome>' },
      { but:   '<state that does NOT change>' },
    ],
  },
}

// ── Story 3 — outline (Specification fidelity, data-driven) ───────────────────

export const VerbNounOutline = {
  story: '<Story Verb–Noun — outline>',
  actor: '<Actor Name>',

  mainFlow: {
    name: '<scenario outline name>',
    steps: [
      { given: 'a **<ConceptA>** *<amount>*' },
      { when:  'the **<actor>** submits with *<method>*' },
      { then:  'the result is *<outcome>*' },
    ],
    examples: [
      { amount: '<value 1>', method: '<method 1>', outcome: '<outcome 1>' },
      { amount: '<value 2>', method: '<method 2>', outcome: '<outcome 2>' },
    ],
  },
}

// ── Story 4 — stub ───────────────────────────────────────────────────────────
// On the story map this is a named card in the sub-epic row — title and actor
// only. No scenarios yet. Appears as a sticky on the wall; nothing to run.

export const VerbNounStub = {
  story: '<Story Verb–Noun — stub>',
  actor: '<Actor Name>',
  // no scenarios — story is on the map but not yet specified
}
